frontend/src/utils/account-send.ts

```diff
@@ -0,0 +1,179 @@
+};
+
+type EncryptAccountDataArgs = KeyData & AccountDataToEncrypt;
+type DecryptAccountDataArgs = KeyData & EncryptedAccountSendData;
+type StoreSendArgs = { chainId: number; viewingKey: string; provider: Web3Provider } & Omit<
+  AccountSendData,
+  'dateSent' | 'recipientId'
+> &
+  Omit<AccountDataToEncrypt, 'address'>;
+type FetchAccountSendArgs = {
+  chainId: number;
+  address: string;
+  viewingKey: string;
+};
+// All values stored in local storage
+type AccountSendDataWithEncryptedFields = UnencryptedAcountSendData & EncryptedAccountSendData;
+// All values in local storage with encrypted values decrypted
+type AccountSendData = UnencryptedAcountSendData & { recipientId: string } & Omit<AccountDataToEncrypt, 'address'>;
+
+export const buildAccountDataForEncryption = ({
+  address,
+  advancedMode,
+  pubKey,
+  usePublicKeyChecked,
+}: AccountDataToEncrypt) => {
+  if (!isAddress(address)) {
+    throw new Error('Invalid address');
+  }
+  if (pubKey.slice(0, 4) !== '0x04') {
+    throw new Error('Invalid public key');
+  }
+
+  address = getAddress(address).slice(2); // slice off the `0x` prefix.
+  const advancedModeHalfByte = advancedMode ? '1' : '0';
+  const usePublicKeyCheckedHalfByte = usePublicKeyChecked ? '1' : '0';
+  const remainingBytes = 11; // 20 bytes for address, 1 byte total for advancedMode and usePublicKeyChecked.
+
+  // Store as much of the pubkey as we can in the remaining bytes. The first 4 characters are `0x04` so we skip those.
+  const pubKeyStart = pubKey.slice(4, remainingBytes * 2 + 4); // Each slot is a half byte meaning remaining bytes must be multiplied by 2
+  const dataToEncrypt = `${address}${advancedModeHalfByte}${usePublicKeyCheckedHalfByte}${pubKeyStart}`;
+  if (dataToEncrypt.length !== 64) {
+    // This should never happen.
+    throw new Error(`Data to encrypt is not the correct length ${dataToEncrypt.length} instead of 64`);
+  }
+
+  return BigNumber.from(`0x${dataToEncrypt}`);
+};
+
+export const encryptAccountData = ({
+  address,
+  advancedMode,
+  usePublicKeyChecked,
+  encryptionCount,
+  viewingKey,
+  pubKey,
+}: EncryptAccountDataArgs) => {
+  const key = keccak256(toUtf8Bytes(`${viewingKey}${encryptionCount}`));
+  const data = buildAccountDataForEncryption({ address, advancedMode, usePublicKeyChecked, pubKey });
+  const encryptedData = data.xor(key);
+  return hexZeroPad(encryptedData.toHexString(), 32);
+};
+
+export const decryptData = ({ viewingKey, encryptionCount, encryptedAddress }: DecryptAccountDataArgs) => {
+  const key = keccak256(toUtf8Bytes(`${viewingKey}${encryptionCount}`));
+
+  const decryptedData = BigNumber.from(encryptedAddress).xor(key);
+  const hexData = decryptedData.toHexString();
+
+  const paritalPubKey = hexData.slice(44);
+  const advancedMode = hexData.slice(42, 43);
+  const usePublicKeyChecked = hexData.slice(43, 44);
+  return {
+    advancedMode,
+    usePublicKeyChecked,
+    address: hexData.slice(0, 42),
+    pubKey: paritalPubKey,
```

Comment:
Nice, I like that! One nit is let's just add a comment above the `type PartialPublicKey` explaining why it's like this, since it's a thing we just made up and it's easier to understand what 0x99 is with a comment then having to find this convo

Reply:
Added a comment here https://github.com/ScopeLift/umbra-protocol/pull/494/commits/1800d456399a3de09a4b2b8014ae450364ad90e2

Reply:
nit but typo in the comment, the bolded word is missing

> We created this type because when storing account send data we only **have** 11 bytes to store

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/cf55badb2206604b4f6a5442a6a5ed9b96084c13